### PR TITLE
Count validation examples

### PR DESCRIPTION
**Pull request by Jsevillamol**
This makes it so the after cuts image metadata is correctly counted when using array mode, introduces a new metadata field containing the validation set break up and creates a utility script to display the metadata after cuts and the validation metadata

### CODE REVIEW

**Comment by aribrill** [comment]
Sorry for the delay in my getting to this.

There's a new metadata filed "num_images_before_cuts_by_tel_and_class_name", but we still have "num_images_after_cuts_by_class_name". Shouldn't the format for before and after cuts be the same?

Otherwise, seems good to me.

File: ctlearn/data_loading.py [hunk 1]
```diff
@@ -375,9 +375,10 @@ def get_metadata(self):
                 'num_selected_telescopes': {tel_type: len(tel_ids) for
                     tel_type, tel_ids in self.selected_telescopes.items()},
                 'num_events_before_cuts_by_class_name': self.num_events_before_cuts_by_class_name,
-                'num_images_before_cuts_by_class_name': self.num_images_before_cuts_by_class_name,
+                'num_images_before_cuts_by_tel_and_class_name': self.num_images_before_cuts_by_tel_and_class_name,
                 'num_events_after_cuts_by_class_name' : self.num_passing_events_by_class_name,
                 'num_images_after_cuts_by_class_name' : self.num_passing_images_by_class_name,
```
Comment (aribrill) [inline ctlearn/data_loading.py]: Shouldn't this be updated to match the new format of the num images before cuts parameter?

File: scripts/print_run_metadata.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    print("\n{} total val examples.".format(total_num_events), file=out)
+    print("Num val examples by particle_id:", file=out)
+    for class_name in num_val_examples_by_class_name:
+        print("{}: {} ({}%)".format( 
+                class_name, 
+                num_val_examples_by_class_name[class_name], 
+                100 * float(num_val_examples_by_class_name[class_name])/total_num_events),
+                file=out
+                )
+
```
Comment (aribrill) [inline scripts/print_run_metadata.py]: It says "no newline at end of file"

**Comment by Jsevillamol** [comment]
> There's a new metadata filed "num_images_before_cuts_by_tel_and_class_name", but we still have "num_images_after_cuts_by_class_name". Shouldn't the format for before and after cuts be the same?

After the loading, we are only picking up the data from one telescope, so it does not really make sense to have statistics per telescope.

We may want to implement this after we implement #54 , but for the moment I changed the name to reflect that they have a different format.

File: ctlearn/data_loading.py (hunk 1 above)
Comment (Jsevillamol) [reply ctlearn/data_loading.py]: To be clear, I have not changed the format - I just renamed the metadata field to better reflect its structure.

File: scripts/print_run_metadata.py (hunk 2 above)
Comment (Jsevillamol) [reply scripts/print_run_metadata.py]: Thanks, I have pushed a new commit. On my version of python it runs smoothly, let me know if it keeps giving you trouble.

**Pull request merged**